webui/react/src/pages/TrialDetails.tsx

```diff
@@ -41,58 +86,87 @@ const TrialDetailsComp: React.FC = () => {
   const experimentId = trial.data?.experimentId;
   const hparams = trial.data?.hparams;
 
-  useEffect(() => {
-    if (!experimentId || !hparams) return;
-    getExperimentDetails({ id: experimentId })
-      .then(experiment => {
-
-        try {
-          const rawConfig = clone(experiment.configRaw);
-          const newDescription = `Continuation of trial ${trialId}, experiment` +
-          ` ${experiment.id} (${rawConfig.description})`;
-          const newSearcher = {
-            max_steps: 100, // TODO add form
-            metric: rawConfig.searcher.metric,
-            name: rawConfig.searcher.name,
-            smaller_is_better: rawConfig.searcher.smaller_is_better,
-            source_trial_id: trialId,
-          };
-          const newHyperparameters = trialHParamsToExperimentHParams(hparams);
-
-          setContModalConfig(yaml.safeDump({
-            ...rawConfig,
-            description: newDescription,
-            hyperparameters: newHyperparameters,
-            searcher: newSearcher,
-          }));
-        } catch (e) {
-          setContModalConfig('failed to load experiment config');
-          handleError({
-            error: e,
-            message: 'failed to load experiment config',
-            type: ErrorType.ApiBadResponse,
-          });
-        }
-      });
-
-  } , [ experimentId, trialId, hparams ]);
-
   const handleActionClick = useCallback((action: TrialAction) => (): void => {
     switch (action) {
       case TrialAction.Continue:
-        setContModalVisible(true);
+        setContFormVisible(true);
         break;
     }
   }, [ ]);
 
-  const [ experiment, setExperiment ] = useState<ExperimentDetails>();
+  const handleContFormVisibile = useCallback((isVisible: boolean) => {
+    return () => setContFormVisible(isVisible);
+  }, []);
+
+  const handleFormFinish = useCallback(({ description, maxLength }) => {
+    setContDescription(description);
+    setContMaxLength(parseInt(maxLength));
+  }, [ ]);
+
+  const handleFormCreate = useCallback(async () => {
+    if (!experiment || !hparams || !trialId) return;
+    const formValues = form.getFieldsValue();
+    handleFormFinish(formValues);
+    try {
+      const expConfig = clone(experiment.configRaw);
+      expConfig.description = formValues.description;
+      setTrialLength(expConfig, parseInt(formValues.maxLength));
```

Comment:
We may also need to shim `min_validation_period` and `min_checkpoint_period` in the same way.

Reply:
looks like `min_validation_period` and `min_checkpoint_period` are the same format as `max_length`? Where they can be `batches`, `epochs` or 'records'?

Reference:
https://github.com/determined-ai/determined/compare/remove-steps

Reply:
yep, they are.

Reply:
would these also apply to experiment forking/creation?